src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -314,44 +319,68 @@ const NetworkAssetSelectInput = ({
     walletAddress,
   ]);
 
-  useEffect(() => {
-    const handleBalanceGet = async () => {
-      if (!sdk || !walletAddress) return;
-      await getSmartWalletBalancesByChain(walletAddress, supportedChains);
-    };
-    handleBalanceGet();
-  }, [supportedChains, sdk, walletAddress]);
-
   const filteredSelectedNetworkAssets = useMemo(() => {
     const filtered = selectedNetworkAssets.filter((asset) => containsText(asset.name, assetSearchQuery) || containsText(asset.symbol, assetSearchQuery));
     return orderBy(filtered, [(asset) => asset.balanceWorthUsd ?? 0, 'name'], ['desc', 'asc']);
   }, [selectedNetworkAssets, assetSearchQuery]);
 
   useEffect(() => {
-    const updateAvalancheBalance = () => {
-      if (!sdk ||
-        !walletAddress ||
-        !preselectedNetwork ||
+    const updateAvalancheSmartWalletBalance = () => {
+      if (!preselectedNetwork ||
         preselectedNetwork.chainId !== CHAIN_ID.AVALANCHE ||
-        !smartWalletBalanceByChain ||
-        filteredSelectedNetworkAssets.length
+        !filteredSelectedNetworkAssets.length ||
+        isLoadingAssets
       ) {
         return;
       }
-      setsmartWalletBalanceByChain([
-        ...smartWalletBalanceByChain,
+      setSmartWalletBalanceByChain((prev) => [
+        ...prev?.filter((element) => element.chain !== CHAIN_ID.AVALANCHE),
         {
           total: sumAssetsBalanceWorth(filteredSelectedNetworkAssets),
-          title: (supportedChains.find(({ chainId }) =>  chainId === CHAIN_ID.AVALANCHE) as Chain).title,
+          title: supportedChains.filter(
+            (element) => element.chainId === CHAIN_ID.AVALANCHE
+          )[0].title,
           chain: CHAIN_ID.AVALANCHE,
         },
       ]);
     };
-    updateAvalancheBalance();
+    updateAvalancheSmartWalletBalance();
   }, [
     sdk,
     supportedChains,
-    walletAddress,
+    accountAddress,
+    preselectedNetwork,
+    filteredSelectedNetworkAssets,
+    isLoadingAssets,
```

Comment:
hmmm adding `isLoadingAsset` might fall into loop, hope you bullet tested this

Reply:
yeah since the function is keeping a check for isLoadingAssets